composer/loggers/wandb_logger.py

```diff
@@ -36,9 +44,39 @@ def state_dict(self) -> StateDict:
         # Storing these fields in the state dict to support run resuming in the future.
         return {"name": wandb.run.name, "project": wandb.run.project, "entity": wandb.run.entity, "id": wandb.run.id}
 
+    def batch_end(self, state: State, logger: Logger) -> None:
+        del logger  # unused
+        # On resnet50, _log_artifacts() caused a 22% throughput degradation
+        # wandb.log_artifact() is async according to the docs
+        # (see https://docs.wandb.ai/guides/artifacts/api#2.-create-an-artifact)
+        # so uploads will not block the training loop
+        # slowdown is likely from extra I/O
+        # Hence, logging every n batches instead of every batch
+        if (state.step + 1) % self.log_artifacts_every_n_batches == 0:
+            self._log_artifacts()
+
+    def epoch_end(self, state: State, logger: Logger) -> None:
```

Comment:
I guess there's no reason we would need `log_artifacts_every_n_epochs` when `1` is a reasonable default?

Reply:
Correct, I figured every epoch is infrequent enough that it's OK to incur the cost triggering wandb then.